app/Classes/Controller/MonksyncController.php

```diff
@@ -0,0 +1,576 @@
+<?php declare(strict_types = 1);
+/**
+ * @author Henrik Gebauer <henrik@mind-hochschul-netzwerk.de>, Ergänzung durch Karsten Hannig <dl1tux@t-online.de>
+ * @license https://creativecommons.org/publicdomain/zero/1.0/ CC0 1.0
+ */
+/* vim-Settings
+ * set mouse-=a
+ * set tabstop=4
+ * set shiftwidth=4
+ * set smarttab
+ */
+
+namespace App\Controller;
+
+use App\Service\Ldap;
+use Hengeb\Router\Attribute\AllowIf;
+use Hengeb\Router\Attribute\Route;
+use Symfony\Component\HttpFoundation\Response;
+
+define("MITNR", "mhnmitnr");
+
+class MonksyncController extends Controller {
+	private $tracelevel = -1;
+	private array $ldapIds =[];
+	private array $ldapUsers =[];
+	private array $monkUsers =[];
+	private array $monkLists =[];
+	private	array $stats=[];
+
+	/* 
+	 * @throws \InvalidArgumentException wrong number of arguments
+	 */ 
+	private function trace_r($tlvl,$lin, $txt,$var) {
+		if(func_num_args() !=4) {
+			throw new \InvalidArgumentException('wrong number of arguments for trace_r called in line '.$lin, __LINE__);
+			self::error(__LINE__,'wrong number of arguments for trace_r called in line '.$lin);
+		}
+		if ($this->tracelevel < $tlvl) return;
+		printf("TRACE_R_%-3d %3d : %s\n", $tlvl, $lin, $txt);
+		print_r($var);
+		printf("\n");
+	}
+
+	/* 
+	 * @throws \InvalidArgumentException wrong number of arguments
+	 */ 
+	private function trace($tlvl,$lin, $txt) {
+		if(func_num_args() !=3) {
+			throw new \InvalidArgumentException('wrong number of arguments for trace_r called in line '.$lin, __LINE__);
+			self::error(__LINE__,'wrong number of arguments for trace called in line '.$lin);
+		}
+		if ($this->tracelevel < 0) {
+			if (empty(getenv("MONKSYNCTRACE"))) {
+				$this->tracelevel = 0;
+			} else {
+				$this->tracelevel = intval(getenv("MONKSYNCTRACE"));
+				printf("TRACE %d : LEVEL: %d\n",__LINE__, $this->tracelevel);
+			}
+		}
+		if ($this->tracelevel < $tlvl) return;
+
+		printf("TRACE_%-3d %3d : %s\n", $tlvl, $lin, $txt);
+	}
+
+	private function error($lin, $txt) {
+		printf("ERROR %d : %s\n", $lin, $txt);
+	}
+
+	private function isMarkedInvalid(String $mail):bool {
+		if ($mail === null
+				|| !str_ends_with(strtolower($mail), ".invalid"))
+			return false;
+		return true;
+	}
+
+	private function stripInvalid(String $mail):String {
+		if (self::isMarkedInvalid($mail) === false)
+			return $mail;
+		return substr($mail, 0, -8);
+	}
+
+	private function searchLdapByMail(String $mail):int {
+		if (empty($mail)) return -1;
```

Comment:
Bitte return in eigene Zeile und geschweifte Klammern einfügen.
Bei den weiteren Vorkommen schreibe ich es nicht dran, da bitte auch ändern.